Спринт 12. Добавление БД

polyakovanan opened the pull request:
Добавил реализацию варианта функционала для работы с БД.
Добавил работу с рейтингами и жанрами.
Привел в соответствие с требованиями функционал списка друзей.

polyakovanan committed:
Спринт 12.
Добавил в проект базу данных, проверил, что поднимается схема и данные справочников и к ней можно подрубиться

polyakovanan committed:
Спринт 12.
Начало работ по БД

polyakovanan committed:
Спринт 12.
Добавил черновые заготовки маппер и репозиториев + БД-стораджи без реализации методов.
Вынес в отдельные класс дружбы и лайки

polyakovanan committed:
Спринт 12.
Разработка основного функционала. Фиксация реализации, которая проходит тестирование на всех кейсах для postman'а

polyakovanan committed:
Спринт 12.
Восстановил работоспособность InMemory реализации с прохождением тест кейсов для postman

polyakovanan committed:
Спринт 12.
Реализовал тесты для БД реализации, покрытие > 87%
Требуется покрыть тестами рейтинги и жанры

polyakovanan committed:
Спринт 12.
Покрытие тестами нового функционала. Исправление багов по результатам тестирования. Исправления по код-стайлу

polyakovanan committed:
Merge branch 'main' into add-database

polyakovanan committed:
Спринт 12.
Увеличил размер поля для email'а по замечаниям из p2p проверке

polyakovanan committed:
Спринт 12.
Увеличил размер поля для email'а по замечаниям из p2p проверке

polyakovanan committed:
Merge remote-tracking branch 'origin/add-database' into add-database

Umka7389 commented on src/main/java/ru/yandex/practicum/filmorate/FilmorateApplication.java:
```diff
@@ -1,9 +1,11 @@
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
 @SpringBootApplication
+@EnableAutoConfiguration
```
SpringBootApplication уже включает аннотацию EnableAutoConfiguration

polyakovanan replied:
Убрал

Umka7389 commented on src/main/java/ru/yandex/practicum/filmorate/storage/film/DbFilmStorage.java:
```diff
@@ -0,0 +1,47 @@
+import java.util.List;
+import java.util.Optional;
+
+@Component("dbFilmStorage")
```
Имя бина автоматический формируется из имени класса в camelCase, в данном случае нет смысла его указывать

polyakovanan replied:
Убрал все подобные дубликаты дефолтных значений

Umka7389 reviewed (commented):
Молодец, хорошая работа! Замечания в комментарии

polyakovanan committed:
Спринт 12. Итерация 1.
Правки по замечаниям к ПР: убрал имена из @Component'ов, которые дублируют дефолтные. Убрал @EnableAutoConfiguration

polyakovanan committed:
Спринт 12. Итерация 1.
Быстрофикс по чек-стайлу

Umka7389 reviewed (commented):
Молодец, хорошая работа!
